operator/v1/types_network.go

```diff
@@ -419,6 +419,64 @@ type OVNKubernetesConfig struct {
 	// egressIPConfig holds the configuration for EgressIP options.
 	// +optional
 	EgressIPConfig EgressIPConfig `json:"egressIPConfig,omitempty"`
+	// ipv4 allows users to configure IP settings for IPv4 connections. When ommitted,
+	// this means no opinions and the default configuration is used. Check individual
+	// fields within ipv4 for details of default values.
+	// +optional
+	IPv4 *IPv4OVNKubernetesConfig `json:"ipv4,omitempty"`
+	// ipv6 allows users to configure IP settings for IPv6 connections. When ommitted,
+	// this means no opinions and the default configuration is used. Check individual
+	// fields within ipv4 for details of default values.
+	// +optional
+	IPv6 *IPv6OVNKubernetesConfig `json;"ipv6,omitempty"`
+}
+
+type IPv4OVNKubernetesConfig struct {
```

Comment:
So what will this struct be used to hold exactly? Like which IP subnets are here versus in other places. For example, the InternalMasqueradeSubnet is under IPv4GatewayConfig. Maybe the type should be named IPv4InternalOVNKubernetesConfig, and all of our internally used subnets are in here (without the Internal prefix on them anymore)?

Reply:
This struct would hold the subnets that we configure internally in ovn-k. The reason that the masqueradeSubnet is under the gatewayConfig is due to where that subnet sits in the ovn-k architecture, but we could simply the api a bit and just have all of our configurable subnets in one place, but I think that since we already introduced that field we would also have to deprecate it following the k8s  guidelines similarly to what we are doing with the join subnet, which would be a bit messy and require some changes to cno

Reply:
OK I dont want to create more work with deprecating the gateway masquerade. I guess this change is fine then.